ares__iface_ip_flags_t cannot has the value 0.

0 is not a possible value for ares__iface_ip_flags_t, but in several places, it takes this value:

1. return 0; (ares__iface_ips.c:257)
2. ares__iface_ip_flags_t      addrflag   = 0; (ares__iface_ips.c:355)
3. ares__iface_ip_flags_t     addrflag = 0; (ares__iface_ips.c:450)

Just tested it, it does warn when you combine 2 enum values as flags:

```
[74/92] Building C object src/lib/CMakeFiles/c-ares.dir/ares_strsplit.c.o
/Users/bhouse/svn/c-ares/src/lib/ares_strsplit.c:97:34: warning: integer constant not in range of enumerated type 'ares__buf_split_t' [-Wassign-enum]
    ARES_BUF_SPLIT_NO_DUPLICATES | ARES_BUF_SPLIT_CASE_INSENSITIVE, &llist);
```

--- a/src/lib/util/ares_iface_ips.c
+++ b/src/lib/util/ares_iface_ips.c
@@ -256,12 +256,12 @@ ares_iface_ip_flags_t ares_iface_ips_get_flags(const ares_iface_ips_t *ips,
   const ares_iface_ip_t *ip;
 
   if (ips == NULL) {
-    return 0;
+    return ARES_IFACE_IP_NONE;
   }
 
   ip = ares_array_at_const(ips->ips, idx);
   if (ip == NULL) {
-    return 0;
+    return ARES_IFACE_IP_NONE;
   }
 
   return ip->flags;
@@ -376,7 +376,7 @@ static ares_status_t ares_iface_ips_enumerate(ares_iface_ips_t *ips,
 
   for (address = addresses; address != NULL; address = address->Next) {
     IP_ADAPTER_UNICAST_ADDRESS *ipaddr     = NULL;
-    ares_iface_ip_flags_t       addrflag   = 0;
+    ares_iface_ip_flags_t       addrflag   = ARES_IFACE_IP_NONE;
     char                        ifname[64] = "";
 
 #  if defined(HAVE_CONVERTINTERFACEINDEXTOLUID) && \
@@ -477,7 +477,7 @@ static ares_status_t ares_iface_ips_enumerate(ares_iface_ips_t *ips,
   }
 
   for (ifa = ifap; ifa != NULL; ifa = ifa->ifa_next) {
-    ares_iface_ip_flags_t addrflag = 0;
+    ares_iface_ip_flags_t addrflag = ARES_IFACE_IP_NONE;
     struct ares_addr      addr;
     unsigned char         netmask  = 0;
     unsigned int          ll_scope = 0;

--- a/src/lib/util/ares_iface_ips.h
+++ b/src/lib/util/ares_iface_ips.h
@@ -28,6 +28,7 @@
 
 /*! Flags for interface ip addresses. */
 typedef enum {
+  ARES_IFACE_IP_NONE      = 0,             /*!< No flags set / unspecified */
   ARES_IFACE_IP_V4 = 1 << 0,        /*!< IPv4 address. During enumeration if
                                      *   this flag is set ARES_IFACE_IP_V6
                                      *   is not, will only enumerate v4

